internal/rbd/rbd_journal.go

```diff
@@ -413,15 +413,19 @@ func updateTopologyConstraints(rbdVol *rbdVolume, rbdSnap *rbdSnapshot) error {
 	var err error
 	if rbdSnap != nil {
 		// check if topology constraints matches snapshot pool
-		rbdVol.Topology, err = util.MatchTopologyForPool(rbdVol.TopologyPools,
+		var poolName string
+		var dataPoolName string
+
+		poolName, dataPoolName, rbdVol.Topology, err = util.MatchPoolAndTopology(rbdVol.TopologyPools,
```

Comment:
instead of returning the poolName and datapoolName from the `MatchPoolAndTopology`. keep the existing code and get the poolName and dataPoolName from `FindPoolAndTopology` directly here? if we do so we don't need any change in `Exists` function above.

Reply:
`MatchPoolAndTopology` and `MatchTopologyForpool` are exactly the same code except for the return parameters.

This function is called twice:
- once from Exists
- once from updateTopologyConstraints

If the function is kept as is, I see two problems.
1. There is code duplication since we have to copy 99% of the function in the updateTopologyContaints
2. util.MatchPoolAndTopology will loose one of its two callers which is defeating the purpose of util module function.

`MatchTopologyForPool` could be a helper function that calls `MatchPoolAndTopology` function and discards the two first result.
Is it really worth for an only user ?

Reply:
Agree. let's keep it as it is.